pipeline.py

```diff
@@ -72,31 +92,118 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+        for article in articles:
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            multiple_tagged_tokens = []
+            for token in tokens:
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+                multiple_tagged_tokens.append(token.get_multiple_tagged())
+            article.save_as(" ".join(cleaned_tokens), kind=ArtifactType.cleaned)
+            article.save_as(" ".join(single_tagged_tokens), kind=ArtifactType.single_tagged)
+            article.save_as(" ".join(multiple_tagged_tokens), kind=ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        processed_text = Mystem().analyze(raw_text)
+        tokens = []
+        pymorph_analyzer = MorphAnalyzer()
+        for processed_word in processed_text:
+            if not processed_word.get('text') or not processed_word.get('analysis'):
+                continue
+            morph_token = MorphologicalToken(original_word=processed_word['text'])
+            morph_token.normalized_form = processed_word['analysis'][0]['lex']
```

Comment:
who guaranteed there will be lex and gr?

Reply:
done